curvefs/src/mds/fs_manager.cpp

```diff
@@ -795,15 +795,22 @@ int FsManager::IsExactlySameOrCreateUnComplete(const std::string& fsName,
         return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
     };
 
-    auto checkFsInfo = [fsType, volumeInfoComparator](const FsDetail& lhs,
-                                                      const FsDetail& rhs) {
+    auto s3InfoComparator = [](common::S3Info lhs, common::S3Info rhs) {
+        // for compatible with old clients
+        lhs.clear_objectprefix();
+        rhs.clear_objectprefix();
```

Comment:
I think we should not compare this directly, but determine whether there is this attribute (has_objectprefix).
1. If not, it means that the request is made by an old client. At this time, the attribute of fs should be required to be 0 (refer to curvefs/conf/tools.conf). If it is other, it should be wrong.
2. If available, just compare directly.

Reply:
there is no need to complicate this logic, the object prefix can't change anymore once create. So we don't need to compare this attribute.

Reply:
> there is no need to complicate this logic, the object prefix can't change anymore once create. So we don't need to compare this attribute.

1. If the relevant attributes of the client and fs are different, it will lead to data inconsistency.
2. The other thing is to be fool-proof. This check can prevent users from using different objectprefix attributes to mount (curveadm's mounting process includes createfs).

Reply:
ok